packages/node/src/handlers/start-coordinator.ts

```diff
@@ -46,30 +46,18 @@ export async function startCoordinator(config: Config) {
   }
 
   // =================================================================
-  // STEP 3: Group unique API calls and validate
+  // STEP 3: Group unique API calls (for the same chain & networkId)
```

Comment:
Does this mean that if we receive requests with the same ID from different chains, we make separate API calls for them and fulfill independently?

Reply:
The implementation looks to me like they would have been aggregated.

Reply:
Yeah good point. The current implementation would group all requests from different chains together if they have the same request ID (so this code comment is wrong).

Would this ever happen in practice though?

Reply:
I think the real question is if it can be used to demonstrate a hypothetical replay attack and the answer is yes. We have two options:
- Don't aggregate requests with identical IDs coming from different chains on the node
- Include the chain ID in the request ID generation like https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md so that request IDs are unique across chains
I think the latter option is much cleaner. It will increase the gas cost a tiny bit and will use a relatively new (Istanbul so not that much) EVM opcode that may require an extra step in porting in some cases. Neither is that important.

Reply:
We don't need to do the same for withdrawals because we don't aggregate them at the node and you can't have the node misbehave (because the main problem is making 1 API call instead of 2)

Reply:
Sure, I'm fine with including the chain ID in the request ID. It does sound a bit better to me too.

Do you want to log a separate issue for it? I'll leave this part of the node as is then (including the code comment)

Reply:
I created the issue (https://github.com/api3dao/airnode/issues/244) and am working on it

I think "Group API calls with respect to request IDs" or something along that line would be the most accurate. This sounds like extra stuff is being done to keep chains/networks separate.